### ISSUE TITLE
DioError [DioErrorType.other]: type 'List<dynamic>' is not a subtype of type 'Map<String, dynamic>?'

### ISSUE THREAD

**Issue by ynnob** [Rexios80/discord_interactions#3]
Hey i tried your example script to run `bulkOverwriteGuildApplicationCommands`

The method will fail:

```
dart run scripts/setup_commands.dart 
DioError [DioErrorType.other]: type 'List<dynamic>' is not a subtype of type 'Map<String, dynamic>?' in type cast
#0      DioMixin.assureResponse (package:dio/src/dio_mixin.dart:818:31)
#1      DioMixin.fetch.<anonymous closure> (package:dio/src/dio_mixin.dart:630:14)
<asynchronous suspension>
#2      DioResponseValidator.validate (package:dio_response_validator/src/dio_response_validator_base.dart:13:18)
<asynchronous suspension>
#3      main (file:///root/XXXXXX/scripts/setup_commands.dart:22:7)
<asynchronous suspension>

Source stack:
#0      DioMixin.fetch (package:dio/src/dio_mixin.dart:488:35)
#1      DioMixin.request (package:dio/src/dio_mixin.dart:483:12)
#2      DioMixin.put (package:dio/src/dio_mixin.dart:139:12)
#3      ApplicationCommandsApi.bulkOverwriteGuildApplicationCommands (package:discord_interactions/src/api/src/interactions/application_commands_api.dart:282:10)
#4      main (file:///root/XXXXXX/scripts/setup_commands.dart:22:37)
#5      _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:297:19)
#6      _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:192:26)

#0      DioMixin.fetch.<anonymous closure> (package:dio/src/dio_mixin.dart:643:7)
#1      _RootZone.runBinary (dart:async/zone.dart:1665:54)
#2      _FutureListener.handleError (dart:async/future_impl.dart:162:22)
#3      Future._propagateToListeners.handleError (dart:async/future_impl.dart:779:47)
#4      Future._propagateToListeners (dart:async/future_impl.dart:800:13)
#5      Future._complete (dart:async/future_impl.dart:558:7)
#6      _SyncCompleter.complete (dart:async/future_impl.dart:47:12)
#7      Future.any.onValue (dart:async/future.dart:613:45)
<asynchronous suspension>
#8      Future.any.onValue (dart:async/future.dart:612:5)
<asynchronous suspension>
```

**Comment by ynnob** [comment]
It looks like this is only triggered by validate() for the returned response.
The actual request did succeed and update the commands.

**Comment by Rexios80** [comment]
Yep I definitely screwed up the migration to using `dio_response_validator`. Every call where the API returns a list was broken. I didn't test it too much because to be honest I thought I was the only one using this. This should be fixed in version `0.0.17` that just released on `pub.dev`. Please let me know if anything else is broken.
